pkg/connection/manager_test.go

```diff
@@ -0,0 +1,369 @@
+/*
+ Copyright 2022 The Crossplane Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package connection
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/connection/fake"
+	"github.com/crossplane/crossplane-runtime/pkg/connection/store"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+)
+
+const (
+	SecretStoreFake v1.SecretStoreType = "Fake"
+
+	fakeConfig = "fake"
+)
+
+const (
+	errBuildStore = "cannot build store"
+)
+
+var (
+	storeConfigGVK = schema.GroupVersionKind{
+		Group:   "testprovider.secrets.crossplane.io",
+		Version: "v1alpha1",
+		Kind:    "StoreConfig",
+	}
+)
+
+func TestManager_connectStore(t *testing.T) {
```

Comment:
This is possibly just a pet peeve of mine, but I'd really prefer us to avoid underscores in test function names - underscores in function names are typically considered unidiomatic. I think for example `TestManagerConnectionStore` would be fine here.

https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md#prefer-table-driven-tests